setup.py

```diff
@@ -45,12 +45,11 @@
         # grpc.Channel.close() method isn't added until 1.32.0.
         # https://github.com/grpc/grpc/pull/15254
         "grpcio >= 1.32.0, < 2.0dev",
-        "pyarrow >= 1.0.0, < 3.0dev",
+        "pyarrow >= 3.0.0, < 4.0dev",
```

Comment:
Can we keep this at 1.0 as a minimum? Or were there breaking changes we needed to account for?

Since pyarrow is a core library, the wider we can keep these dependencies, the fewer conflicts we'll get.